test/unit/community-voice-logic.spec.ts

```diff
@@ -0,0 +1,255 @@
+    } as jest.Mocked<ILoggerComponent>
+
+    mockAnalytics = {
+      fireEvent: jest.fn(),
+      sendEvent: jest.fn()
+    } as jest.Mocked<IAnalyticsComponent<AnalyticsEventPayload>>
+
+    voiceComponent = createVoiceComponent({
+      voiceDB: mockVoiceDB,
+      livekit: mockLivekit,
+      logs: mockLogs,
+      analytics: mockAnalytics
+    })
+  })
+
+  describe('getCommunityVoiceChatCredentialsForModerator', () => {
+    beforeEach(() => {
+      mockLivekit.getRoom.mockResolvedValue({
+        name: 'voice-chat-community-test-community-123',
+        sid: 'test-sid',
+        emptyTimeout: 10,
+        departureTimeout: 10,
+        maxParticipants: 100,
+        creationTime: Date.now(),
+        enabledCodecs: [],
+        metadata: '',
+        numParticipants: 0,
+        numPublishers: 0,
+        activeRecording: false
+      } as any) // Using any to avoid importing full Room type
+      mockLivekit.generateCredentials.mockResolvedValue({
+        url: 'wss://voice.livekit.cloud',
+        token: 'moderator-token'
+      })
+      mockLivekit.buildConnectionUrl.mockReturnValue('livekit:wss://voice.livekit.cloud?access_token=moderator-token')
+    })
+
+    it('should generate credentials for moderator with correct permissions', async () => {
+      const result = await voiceComponent.getCommunityVoiceChatCredentialsForModerator(
+        validCommunityId,
+        validModeratorAddress
+      )
+
+      expect(result).toEqual({
+        connectionUrl: 'livekit:wss://voice.livekit.cloud?access_token=moderator-token'
+      })
+
+      expect(mockLivekit.getRoom).toHaveBeenCalledWith(getCommunityVoiceChatRoomName(validCommunityId))
+
+      expect(mockLivekit.generateCredentials).toHaveBeenCalledWith(
+        validModeratorAddress,
+        getCommunityVoiceChatRoomName(validCommunityId),
+        {
+          cast: [],
+          canPublish: true,
+          canSubscribe: true,
+          canUpdateOwnMetadata: false
+        },
+        false,
+        {
+          role: CommunityRole.Moderator,
+          community_id: validCommunityId
+        }
+      )
+
+      expect(mockVoiceDB.joinUserToCommunityRoom).toHaveBeenCalledWith(
+        validModeratorAddress,
+        getCommunityVoiceChatRoomName(validCommunityId),
+        true
+      )
+    })
+
+    it('should insert moderator as NotConnected in database (not Connected)', async () => {
+      await voiceComponent.getCommunityVoiceChatCredentialsForModerator(validCommunityId, validModeratorAddress)
+
+      // Verify that joinUserToCommunityRoom is called to insert user in DB
+      // The actual implementation will insert with NotConnected status
+      expect(mockVoiceDB.joinUserToCommunityRoom).toHaveBeenCalledWith(
+        validModeratorAddress,
+        getCommunityVoiceChatRoomName(validCommunityId),
+        true // isModerator = true
+      )
+    })
+
+    it('should handle livekit errors properly', async () => {
```

Comment:
As this test follows a different context, that is, the context where LiveKit fails to get a room, per the testing standard, this should be described in a different describe. Would you mind changing that?
What do you think about changing the it text of the test to: `should propagate LiveKit errors`?